cluster/saltbase/salt/kube-addons/kube-addons.sh

```diff
@@ -134,6 +134,15 @@ for k,v in yaml.load(sys.stdin).iteritems():
 ''' < "${kube_env_yaml}")
 fi
 
+# Wait for the default service account
+token_found=""
+while [ -z "${token_found}" ]; do
```

Comment:
At least generate a horrible-looking error message here if you're retried more than a reasonable number of times, and return non-zero from this script.  At least that way we get an explicit error with details of what went wrong, rahter than a silently not working cluster.